### PR TITLE
#[derive_no_bound] and #[structural_match]

### PR DESCRIPTION
**Pull request by Centril**
Patching the RFC wrt. `#[structural_match]`.

### CODE REVIEW

**Comment by Centril** [comment]
Review ping @nox.

File: text/0000-derive-bound-control.md [hunk 1]
```diff
@@ -570,6 +570,71 @@ Deriving any standard library trait will obey the semantics here specified.
...
+the type are also `#[structural_match]`.
+
+To use `const` as the pattern in a match arm, it has to be of a type that is
+`#[structural_match]`. If it is not, as in the example below:
+
+```rust
+fn main() {
+    use std::marker::PhantomData;
+
+    pub const PD: PhantomData<u8> = PhantomData;
```
Comment (nox) [inline text/0000-derive-bound-control.md]: `BOO` is cuter.

File: text/0000-derive-bound-control.md [hunk 2]
```diff
@@ -570,6 +570,71 @@ Deriving any standard library trait will obey the semantics here specified.
...
+#[derive(PartialEq, Eq, ...)]
+#[derive_no_bound]
+#[lang_item = "phantom_data"]
+struct PhantomData<T: ?Sized>;
+```
+
+With the new definition in (2), the error above will not be emitted and the
+program will be accepted.
+
+This change is does not move us from structural matching. A `PhantomData<T>`
```
Comment (nox) [inline text/0000-derive-bound-control.md]: The "is" looks like a mistake here.

File: text/0000-derive-bound-control.md [hunk 3]
```diff
@@ -570,6 +570,71 @@ Deriving any standard library trait will obey the semantics here specified.
...
+2. `PhantomData<T>` will be defined as:
+
+```rust
+#[derive(PartialEq, Eq, ...)]
+#[derive_no_bound]
+#[lang_item = "phantom_data"]
+struct PhantomData<T: ?Sized>;
+```
+
+With the new definition in (2), the error above will not be emitted and the
```
Comment (nox) [inline text/0000-derive-bound-control.md]: "With this new definition"?

**Review by nox — APPROVED** [review]
A few nits, but otherwise LGTM.

File: text/0000-derive-bound-control.md (hunk 1 above)
Comment (Centril) [reply text/0000-derive-bound-control.md]: 👻 says 😆

**Comment by Centril** [comment]
Fixed the nits, thanks.

**Pull request merged**